### PR TITLE
Fix/abis handle contracts 08

### PR DESCRIPTION
**Pull request by nicolasbrugneaux**
Handles the new contracts folder (build/contracts, build/contracts-08, build/contracts-mento) in a nicer way. Refactors the published package to support both esm and commonjs modules.

This PR also changes the published abis to be ONLY core contracts as it's what devs will want to interact with, and not random internal abis (eg: Math.json)

### CODE REVIEW

File: packages/protocol/abis/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "compileOnSave": false,
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "baseUrl": ".",
+    "jsx": "preserve",
```
Comment (aaronmgdr) [inline packages/protocol/abis/tsconfig.json]: this isn't needed. as this is not a react app

File: packages/protocol/scripts/prepare-contracts-and-abis-publishing.ts [hunk 2]
```diff
@@ -2,51 +2,96 @@ import * as child_process from 'child_process'
...
+  `BUILD_DIR=./build/abis/src ts-node ${BUILD_EXECUTABLE} --coreContractsOnly --web3Types ${path.join(
+    BUILD_DIR,
+    'web3'
+  )}`,
+  {
+    stdio: 'inherit',
+  }
 )
 
+// Merge contracts-0.8, contracts-mento, etc.. at the root of the build dir
```
Comment (aaronmgdr) [inline packages/protocol/scripts/prepare-contracts-and-abis-publishing.ts]: very grateful for these comments

File: packages/protocol/abis/package.json [hunk 3]
```diff
@@ -9,6 +9,10 @@
     "directory": "packages/protocol/contracts"
   },
   "scripts": {},
-  "dependencies": {},
-  "types": "./types/ethers/index.d.ts"
+  "dependencies": {},  
+  "main": "./lib/cjs/index.js",
+  "module": "./lib/esm/index.js",
```
Comment (aaronmgdr) [inline packages/protocol/abis/package.json:14]: why is types not needed anymore?

Comment (nicolasbrugneaux) [reply packages/protocol/abis/package.json:14]: oh shoot

**Review by aaronmgdr — APPROVED** [review]

**Review by aaronmgdr — CHANGES_REQUESTED** [review]
soorry

**Comment by aaronmgdr** [comment]
sorry i want to release packages before making big update to master

**Review by aaronmgdr — APPROVED** [review]
ok go

**Pull request merged**